### ISSUE TITLE
Keyboard of Kindle DXG stops working

### ISSUE THREAD

**Issue by Hzj-jie** [koreader/koreader#12247]
* KOReader version: 2024.07 "Outside"
* Device: Kindle DXG

#### Issue

Keyboard stops working in the latest released version, last known good is 2024.04. Only the power button works.

#### Steps to reproduce

Currently my device is running in reading mode, not very sure if it also applies to the file manager.

##### `crash.log` (if applicable)

```
ffi.load: libs/libutf8proc.so.3
com.lab126.blanket failed to set value for property unload (0x3)
ffi.load: blitbuffer
11/11/23-20:43:01 WARN  could not load liblipclua: module 'liblipclua' not found:
	no field package.preload['liblipclua']
	no file 'common/liblipclua.lua'
	no file 'frontend/liblipclua.lua'
	no file './liblipclua.lua'
	no file '/usr/local/share/luajit-2.1/liblipclua.lua'
	no file '/usr/local/share/lua/5.1/liblipclua.lua'
	no file '/usr/local/share/lua/5.1/liblipclua/init.lua'
	no file 'common/liblipclua.so'
	no file 'common/liblipclua.dll'
	no file '/usr/lib/lua/liblipclua.so'
	no file './liblipclua.so'
	no file '/usr/local/lib/lua/5.1/liblipclua.so'
	no file '/usr/local/lib/lua/5.1/loadall.so' 
ffi.load: fbink_input
ffi.load (assisted searchpath): ./libs/libfbink_input.so.1
[FBInk] /dev/input/event0: `mxckpd` = KEY | MENU_BUTTON
[FBInk] /dev/input/event1: `fiveway` = KEY
[ko-input] Forked off fake event generator (pid: 4951)
11/11/23-20:43:01 INFO  initializing for device KindleDXG 
11/11/23-20:43:01 INFO  framebuffer resolution: {
  h = 1200,
  w = 824
} --[[table: 0x405f1ee0]] 
ffi.load: libs/libwrap-mupdf.so
ffi.load: sqlite3
ffi.load: libs/libfreetype.so.6
ffi.load: libs/libharfbuzz.so.0
ffi.load: libs/libzstd.so.1
11/11/23-20:43:10 INFO  opening file /mnt/us/documents/几何原本.pdf 
ffi.load: libs/libleptonica.so.6
ffi.load: libs/libk2pdfopt.so.2
11/11/23-20:43:16 INFO  Inhibiting user input 
11/11/23-20:43:16 INFO  Loading plugins from directory: plugins 
sh: command: not found
sh: command: not found
11/11/23-20:43:22 INFO  setting zoom mode to content 
11/11/23-20:43:44 INFO  Restoring user input handling 
```

**Comment by benoit-pierre** [comment]
Duplicate of #12178 maybe? Have you tried the nightly?

**Comment by NiLuJe** [comment]
The fiveway certainly wouldn't work, yeah.

I don't see anything flagged with pagination buttons, though, as that's rather more concerning ;p.

(The lack of keyboard flag is okay: one, we don't use it, two, it requires a full set of ESC + 1-0 + Q-D  (ie., a keyboard ;D), so none of the Kindles would qualify).

Can you post the output of `evemu-record /dev/input/event0` & `evemu-record /dev/input/event1` (or evtest, but I fear forthe verbosity of whatever hilariously old version might be shipped on stock FW. Decent versions of both ship w/ USBNet, FWIW).

**Comment by NiLuJe** [comment]
o, much like in #12178, does unpacking that in `libs` to replace `libfbink_input.so.1` fix it?

[fix-fw2-pagination.tar.gz](https://github.com/user-attachments/files/16448418/fix-fw2-pagination.tar.gz)

**Comment by Hzj-jie** [comment]
Sorry for the late response, I will take a shot later.

**Comment by NiLuJe** [comment]
Likely to be fixed via https://github.com/koreader/koreader-base/pull/1896 in #12311, TBC as my only FW 2.x device died a horrible death (dead battery seems to be really dead and bloated, no time (or spare parts) to open it to confirm).

**Comment by Commodore64user** [comment]
> my only FW 2.x device died a horrible death (dead battery seems to be really dead and bloated, no time (or spare parts) to open it to confirm). 

FWIW is just a few screws https://youtu.be/0vqUvf6_qYQ and replacement batteries can be found on eBay or AliExpress.

**Comment by Hzj-jie** [comment]
Sorry, I still haven't got a chance to try. No wifi is actually a noticeable inconvenience.

**Comment by zaviazun** [comment]
I share the same problem as Hzj-jie. The power button is the only one that functions in koreader-kindle-legacy-v2024.07. And than I changed file  libfbink_input.so.1 in the libs as NiLuJe recomended. Thus, the joystick and power button are now functioning properly. Other buttons are inoperable. Please assist in creating the proper "libfbink_input.so.1" file for the Kindle DXG!

**Comment by benoit-pierre** [comment]
Did you test the nightly?

**Comment by zaviazun** [comment]
No, I didn't. Only the koreader-kindle-legacy-v2024.07 bild was tested by me. However, I could try "nightly bild" immediately. Would you kindly provide the URL?

**Comment by benoit-pierre** [comment]
https://build.koreader.rocks/download/nightly/v2024.07-181-gbf290e0c2_2024-09-14/

**Comment by zaviazun** [comment]
The "koreader-kindle-legacy-v2024.07-181-gbf290e0c2_2024-09-14" build was installed, and the Kindle DXG buttons functions as it should. Many thanks!

**Comment by NiLuJe** [comment]
Fixed in #12311 (c.f., https://github.com/koreader/koreader/issues/12247#issuecomment-2274857155).

Thanks for testing this ;).
